platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformBuilder.java

```diff
@@ -0,0 +1,375 @@
+
+        // Setup logging, using the configuration to tell us the location of the log4j2.xml
+        final PathsConfig pathsConfig = configuration.getConfigData(PathsConfig.class);
+        final Path log4jPath = pathsConfig.getLogPath();
+        try {
+            Log4jSetup.startLoggingFramework(log4jPath).await();
+        } catch (final InterruptedException e) {
+            CommonUtils.tellUserConsole("Interrupted while waiting for log4j to initialize");
+            Thread.currentThread().interrupt();
+        }
+
+        // Now that we have a logger, we can start using it for further messages
+        logger.info(STARTUP.getMarker(), "\n\n" + STARTUP_MESSAGE + "\n");
+        logger.debug(STARTUP.getMarker(), () -> new NodeStartPayload().toString());
+
+        BootstrapUtils.performHealthChecks(configPath, configuration);
+        writeSettingsUsed(configuration);
+
+        metricsProvider = new DefaultMetricsProvider(configuration);
+        globalMetrics = metricsProvider.createGlobalMetrics();
+        CryptoMetrics.registerMetrics(globalMetrics);
+
+        // Initialize the thread dump generator, if enabled via settings
+        startThreadDumpGenerator(configuration);
+
+        // Initialize JVMPauseDetectorThread, if enabled via settings
+        startJVMPauseDetectorThread(configuration);
+
+        return true;
+    }
+
+    /**
+     * Build the configuration for the node.
+     *
+     * @return the configuration
+     */
+    @NonNull
+    private Configuration buildConfiguration() {
+        if (configurationBuilder == null) {
+            configurationBuilder = ConfigurationBuilder.create();
+        }
+
+        if (!Files.exists(configPath)) {
+            throw new IllegalStateException("File " + configPath + " does not exist");
+        }
+        if (!Files.exists(settingsPath)) {
+            throw new IllegalStateException("File " + settingsPath + " does not exist");
+        }
+
+        ConfigUtils.scanAndRegisterAllConfigTypes(configurationBuilder, Set.of(SWIRLDS_PACKAGE));
+        rethrowIO(() -> BootstrapUtils.setupConfigBuilder(configurationBuilder, configPath, settingsPath));
+
+        final Configuration configuration = configurationBuilder.build();
+        PlatformConfigUtils.checkConfiguration(configuration);
+
+        return configuration;
+    }
+
+    /**
+     * Parse the address book from the config.txt file.
+     *
+     * @return the address book
+     */
+    private AddressBook loadConfigAddressBook() {
+        final LegacyConfigProperties legacyConfig = LegacyConfigPropertiesLoader.loadConfigFile(configPath);
+        legacyConfig.appConfig().ifPresent(c -> ParameterProvider.getInstance().setParameters(c.params()));
+        return legacyConfig.getAddressBook();
+    }
+
+    /**
+     * Build a platform.
+     *
+     * @param start if true then start the platform
+     * @return a new platform instance
+     */
+    public SwirldsPlatform build(final boolean start) {
```

Comment:
Why? Can't we just define the builder to return a platform that is never started? That would make it way more easy from my point of view.

Reply:
Mostly just for convenience, but since you don't like it I will remove it.